File: stack/simplify_path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        stack = []
        path_list = path.split('/')

        for p in path_list:
            if stack and p == '..':
                stack.pop()
            elif p not in ['','.','..']:
                stack.append(p)

        return '/'+ '/'.join(stack)

File: stack/simplify_path.java
package stack;

import java.util.Stack;

public class simplify_path {
    public String simplifyPath(String path) {
        Stack<String> stk = new Stack<>();
        String sr[] = path.split("/");
        for(String s: sr){
            if(stk.size() != 0 && s.equals("..")){
                stk.pop();
            }
            else if(!(s.equals("") || s.equals(".") || s.equals(".."))){
                stk.push(s);
            }
        }
        return "/"+String.join("/",stk);
    }
}
